qiskit/backends/ibmq/ibmqprovider.py

```diff
@@ -5,50 +5,56 @@
 # This source code is licensed under the Apache License, Version 2.0 found in
 # the LICENSE.txt file in the root directory of this source tree.
 
-"""Provider for remote IbmQ backends."""
-from IBMQuantumExperience import IBMQuantumExperience
+"""Provider for remote IBMQ backends with admin features."""
+from collections import OrderedDict
 
-from qiskit._util import _camel_case_to_snake_case
-from qiskit.backends.baseprovider import BaseProvider
-from qiskit.backends.ibmq.ibmqbackend import IBMQBackend
-from qiskit._util import _parse_ibmq_credentials
+from qiskit import QISKitError
 
+from qiskit.backends import BaseProvider
 
-class IBMQProvider(BaseProvider):
-    """Provider for remote IbmQ backends."""
-    def __init__(self, token, url='https://quantumexperience.ng.bluemix.net/api',
-                 hub=None, group=None, project=None, proxies=None, verify=True):
-        super().__init__()
-
-        url = _parse_ibmq_credentials(url, hub, group, project)
+from .credentials._configrc import remove_credentials
+from .credentials import (Credentials,
+                          read_credentials_from_qiskitrc, store_credentials, discover_credentials)
+from .ibmqsingleprovider import IBMQSingleProvider
 
-        # Get a connection to IBMQuantumExperience.
-        self._api = self._authenticate(token, url, proxies=proxies, verify=verify)
+QE_URL = 'https://quantumexperience.ng.bluemix.net/api'
 
-        # Populate the list of remote backends.
-        self.backends = self._discover_remote_backends()
 
-        # authentication attributes, which uniquely identify the provider instance
-        self._token = token
-        self._url = url
-        self._proxies = proxies
-        self._verify = verify
-
-    def get_backend(self, name):
-        return self.backends[name]
+class IBMQProvider(BaseProvider):
+    """Provider for remote IBMQ backends with admin features."""
+    def __init__(self):
```

Comment:
Would it make sense to add an optional credentials arg here so you can initialize a provider with credentials objects when you create it?

Reply:
Hmm, since this class is primarily going to be instantiated "automagically" (and by the time it is instantiated, we don't really know any credentials) and is the "admin" class, I'm not sure we would gain too much compared to the complexity it would introduced, right?